src/field_theory/cardinality.lean

```diff
@@ -69,3 +70,16 @@ begin
     simpa [mv_polynomial.monomial_eq_monomial_iff, finsupp.single_eq_single_iff] using h },
   { simpa using @mv_polynomial.cardinal_mk_le_max α (ulift.{u} ℚ) _ }
 end
+
+/-- There is a field structure on type if its cardinality is a prime power. -/
+lemma field.nonempty_iff {α : Type u} : nonempty (field α) ↔ is_prime_pow (#α) :=
+begin
+  rw cardinal.is_prime_pow_iff,
+  by_cases ω ≤ (#α),
+  { simp only [h, true_or, iff_true],
+    haveI := cardinal.infinite_iff.mpr h,
+    exact infinite.nonempty_field },
+  casesI cardinal.lt_omega_iff_fintype.mp (not_le.mp h),
+  simp_rw [h, false_or, cardinal.mk_fintype, nat.cast_inj, exists_eq_left'],
+  exact fintype.nonempty_field_iff
+end
```

Comment:
I agree with Violeta that the `by_cases` seems awkward. How about `fintype_or_infinite`?
```suggestion
begin
  rw cardinal.is_prime_pow_iff,
  casesI fintype_or_infinite α with h h,
  { simpa only [cardinal.mk_fintype, nat.cast_inj, exists_eq_left',
        (cardinal.nat_lt_omega _).not_le, false_or]
      using fintype.nonempty_field_iff },
  { simpa only [← cardinal.infinite_iff, h, true_or, iff_true]
      using infinite.nonempty_field },
end
```